[go-server] Null Pointer exception when request body schema is array

<!--
Please follow the issue template below for bug reports and feature requests.
Also please indicate in the issue title which language/library is concerned. Eg:  [JAVA] Bug generating foo with bar 
-->

##### Description

<!-- describe what is the question, suggestion or issue and why this is a problem for you. -->
When the body of a request is an array NPE occurs

```java
  Exception: null
	at org.openapitools.codegen.DefaultGenerator.processOperation(DefaultGenerator.java:934)
	at org.openapitools.codegen.DefaultGenerator.processPaths(DefaultGenerator.java:828)
	at org.openapitools.codegen.DefaultGenerator.generateApis(DefaultGenerator.java:460)
	at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:786)
	at org.openapitools.codegen.cmd.Generate.run(Generate.java:315)
	at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:58)
Caused by: java.lang.NullPointerException
	at org.openapitools.codegen.languages.AbstractGoCodegen.toVarName(AbstractGoCodegen.java:148)
	at org.openapitools.codegen.languages.AbstractGoCodegen.toParamName(AbstractGoCodegen.java:181)
	at org.openapitools.codegen.DefaultCodegen.toArrayModelParamName(DefaultCodegen.java:760)
	at org.openapitools.codegen.DefaultCodegen.fromRequestBody(DefaultCodegen.java:4214)
	at org.openapitools.codegen.DefaultCodegen.fromOperation(DefaultCodegen.java:2198)
	at org.openapitools.codegen.DefaultGenerator.processOperation(DefaultGenerator.java:902)
	... 5 more
```

##### openapi-generator version

<!-- which version of openapi-generator are you using, is it a regression? -->
`master`
##### OpenAPI declaration file content or url

<!-- if it is a bug, a json or yaml that produces it.
If you post the code inline, please wrap it with
```yaml
(here your code)
```
(for YAML code) or
```json
(here your code)
```
(for JSON code), so it becomes more readable. If it is longer than about ten lines,
please create a Gist (https://gist.github.com) or upload it somewhere else and
link it here.
  -->
```yaml
---
# This is a sample Swagger spec
 
swagger: "2.0"
info:
  description: Container Application Model Description
  version: 1.0.0
  title: ContainerApplication
  
host: localhost:32000
basePath: /api/v1.0
schemes:
- http

tags:  
- name: ApplicationEndpointTest
  x-displayName: "ApplicationEndpointTest"
  description: |
    "Test of endpoints "
    
paths:
  /default/EndpointTest: 
    post:
      tags:
      - ApplicationEndpointTest      
      description: "post a request"
      operationId: callEndpoint
      consumes:
      - application/json
      parameters:
      - name: "requestList"
        in: "body"
        description: "Requests to post"
        required: true
        schema:
          $ref: "#/definitions/listitems"
      responses:
        200:
          description: "Successful response"
        404:
          description: "The request was not successfully executed."
      
  
definitions:
  listitems : 
    description: "list of items"
    type: array
    items: 
      type: "string"
```
##### Command line used for generation

<!-- including the language, libraries and various options -->
java -jar ~/openapi-generator/modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -i ./api.yaml -o ./gengo-server -g go-server

##### Steps to reproduce

<!-- unambiguous set of steps to reproduce the bug.-->

##### Related issues/PRs

<!-- has a similar issue/PR been reported/opened before? Please do a search in https://github.com/openapitools/openapi-generator/issues?utf8=%E2%9C%93&q=is%3Aissue%20 -->

##### Suggest a fix/enhancement

<!-- if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit), or simply make a suggestion -->

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java
@@ -144,8 +144,12 @@ public abstract class AbstractGoCodegen extends DefaultCodegen implements Codege
 
     @Override
     public String toVarName(String name) {
+    	
+    	if(name != null) {
+    		name = name.replaceAll("-", "_");
+    	}
         // replace - with _ e.g. created-at => created_at
-        name = sanitizeName(name.replaceAll("-", "_"));
+        name = sanitizeName(name);
 
         // if it's all uppper case, do nothing
         if (name.matches("^[A-Z_]*$"))

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java
@@ -144,7 +144,7 @@ public abstract class AbstractGoCodegen extends DefaultCodegen implements Codege
 
     @Override
     public String toVarName(String name) {
-    	
+        
         // replace - with _ e.g. created-at => created_at
         name = sanitizeName(name);
 

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java
@@ -145,9 +145,6 @@ public abstract class AbstractGoCodegen extends DefaultCodegen implements Codege
     @Override
     public String toVarName(String name) {
     	
-    	if(name != null) {
-    		name = name.replaceAll("-", "_");
-    	}
         // replace - with _ e.g. created-at => created_at
         name = sanitizeName(name);
 

